### PR TITLE
docs(lsp): update an outdated example and mark optional parameters

### PR DESCRIPTION
**Pull request by ofseed**
## Problem:
* bufnr is actually optional in `Client:request_sync`
* `tsserver` was renamed to `ts_ls` in `nvim-lspconfig`

## Solution:
Fix them.

### CODE REVIEW

File: runtime/lua/vim/lsp/client.lua [hunk 1]
```diff
@@ -715,7 +715,7 @@ end
 --- @param params table LSP request params.
 --- @param timeout_ms integer? Maximum time in milliseconds to wait for
 ---                                a result. Defaults to 1000
---- @param bufnr integer Buffer handle (0 for current).
+--- @param bufnr? integer Buffer handle (0 for current).
```
Comment (MariaSolOs) [inline runtime/lua/vim/lsp/client.lua:718]: You need to run `make doc` to add this to the respective text file

Comment (ofseed) [reply runtime/lua/vim/lsp/client.lua:718]: Sorry, I forgot and force-pushed now.

**Comment by MariaSolOs** [comment]
Despite the straightforward changes, a non-empty PR description and a title following [conventional commit guidelines](https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md#commit-messages) are always appreciated :)

File: runtime/lua/vim/lsp/client.lua (hunk 1 above)
Comment (MariaSolOs) [inline runtime/lua/vim/lsp/client.lua:718]: Could you also mention that `bufnr` defaults to 0 please?

**Comment by ofseed** [comment]
> > Despite the straightforward changes, a non-empty PR description and a title following [conventional commit guidelines](https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md#commit-messages) are always appreciated :)

I'm not very sure about this, because I've seen that some minor documentation changes are collected into a single PR for merging, so I'm currently leaving the PR for such a multifaceted change empty, with meaningful commit messages only. but I will do this.

**Comment by MariaSolOs** [comment]
> I'm not very sure about this, because I've seen that some minor documentation changes are collected into a single PR for merging, so I'm currently leaving the PR for such a multifaceted change empty, with meaningful commit messages only. but I will do this.

Those kind of guidelines aren't only for changelog stuff, but also to make it easier for reviewers and a cleaner GitHub history!

File: runtime/lua/vim/lsp/client.lua (hunk 1 above)
Comment (justinmk) [inline runtime/lua/vim/lsp/client.lua:718]:
```suggestion
--- @param bufnr? integer (default: 0) Buffer handle, or 0 for current.
```

Comment (ofseed) [reply runtime/lua/vim/lsp/client.lua]: The original purpose here is to be the same as the `Client:request` above, so I have pushed a change that simultaneously improves both places now.

File: runtime/lua/vim/lsp/client.lua [hunk 2]
```diff
@@ -715,7 +715,7 @@ end
 --- @param params table LSP request params.
 --- @param timeout_ms integer? Maximum time in milliseconds to wait for
 ---                                a result. Defaults to 1000
---- @param bufnr integer Buffer handle (0 for current).
+--- @param bufnr? integer integer (default: 0) Buffer handle, or 0 for current.
```
[the remark this answers is not in the document]
Comment (ofseed) [reply runtime/lua/vim/lsp/client.lua]: Oops🫠

**Review by justinmk — APPROVED** [review]
CI failure

**Pull request merged**